kb/prometheus-grafana-setup.md

```diff
@@ -0,0 +1,289 @@
+
+For example, in the modified local docker-compose Temporal Cluster setup described in the previous section, create a separate container with port 8085 for Grafana.
+
+```
+version: "3.5"
+services:
+#...
+  grafana:
+   container_name: grafana
+   image: grafana/grafana:7.5.16
+   environment:
+     - GF_AUTH_DISABLE_LOGIN_FORM=true
+     - GF_AUTH_ANONYMOUS_ENABLED=true
+     - GF_AUTH_ANONYMOUS_ORG_ROLE=Admin
+   ports:
+     - 8085:3000
+   volumes:
+     - type: bind
+       source: ./deployment/grafana/provisioning/datasources
+       target: /etc/grafana/provisioning/datasources
+   depends_on:
+     - prometheus
+#...
+```
+
+Note that in this example, Grafana is set up to start without authorizations; this is not a good practice and is not recommended.
+For more information on how to custimize your Grafana setup, see the [Grafana documentation](https://grafana.com/docs/grafana/latest/setup-grafana/).
+
+Next, configure Grafana to use Prometheus as the datasource. You can do this either on the UI or in your Grafana deployment configuration.
+
+For the example above, create a configuration file (for example - config.yml) file at ./deployment/grafana/provisioning/datasource in your docker-compose Temporal Cluster project and configure the Prometheus data source for Grafana, as shown:
+
+```
+apiVersion: 1
+
+datasources:
+ - name: 'Temporal Prometheus'
+   type: 'prometheus'
+   org_id: 1
+   url: 'http://prometheus:9090'
+   is_default: true
+   version: 1
+   editable: true
+```
+
+In this example, Grafana is set to pull metrics from Prometheus at the port 9090, as defined in the Prometheus configuration.
+After you update this configuration, restart your local docker-compose Temporal Cluster, and go to [localhost:8085](http://localhost:8085) to access Grafana.
+
+### Grafana dashboard setup
+
+Set up your dashboards in Grafana, using either the UI or configure your dashboards in your Grafana deployment on the Cluster, as done in this [example](https://github.com/tsurdilo/my-temporal-dockercompose/tree/main/deployment/grafana/dashboards).
+
+In your Grafana dashboard, add your Prometheus query to call specific metrics.
+The [Temporal Cluster Metrics reference](/references/cluster-metrics) describes a few metrics and queries that you can get started with.
+
+For example, to create a dashboard in your local Grafana UI at [localhost:8085](http://localhost:8085):
+
+1. Go to Create > Dashboard, and add an empty panel.
+2. On the Panel configuration page, in the Query tab, select "Temporal Prometheus" as the datasource.
+3. In the Metrics field, copy any of the queries listed in the [Temporal Cluster Metrics reference](/references/cluster-metrics);
+   For example, add `sum by (operation) (rate(service_requests{service_name="frontend"}[2m]))` to see all the Frontend Service requests on your local docker-compose Temporal Cluster.
+4. You should see the graph show metrics data for the Frontend Service from the docker-compose Temporal Cluster.
```

Comment:
What does one do with the query copied in the preceding step?

Reply:
good point; you paste the Query in the metrics field to start seeing data in the graph. updated this to:
"In the Metrics field, use any of the queries listed in the [Temporal Cluster Metrics reference](/references/cluster-metrics);"